python/src/trezorlib/debuglink.py

```diff
@@ -374,28 +407,35 @@ def __init__(self, transport, auto_interact=True):
 
         super().__init__(transport, ui=self.ui)
```

Comment:
try moving `self.reset_debug_features()` after this line, that's my only guess why mypy would infer `self.ui` to be of type `DebugUI`. We need to somehow declare that `self.ui` is of type `TrezorClientUI`

Reply:
Actually the `DebugUI` was defined as a type of `ui` for `TrezorClient`. I made it on purpose, because it was the only one having some properties being used on `TrezorClientDebugLink` (`self.ui.pins` or `self.ui.passphrase`)

After setting `TrezorClient.ui: TrezorClientUI` and `TrezorClientDebugLink.ui: DebugUI` it is not complaining anymore (which could be little strange, as `TrezorClientDebugLink` is inheriting from `TrezorClient` - seems children can legally change the type of variable - and it is not even complaining that we input `DebugUI` into it in `super().__init__(transport, ui=self.ui)`)